test/test_videoapi.py

```diff
@@ -0,0 +1,213 @@
+            self.assertAlmostEqual(
+                config.video_fps, reader_md["video"]["fps"][0], delta=0.0001
+            )
+            self.assertAlmostEqual(
+                config.duration, reader_md["video"]["duration"][0], delta=0.5
+            )
+
+    def test_frame_reading(self):
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+
+            av_reader = av.open(full_path)
+
+            if av_reader.streams.video:
+                video_reader = VideoReader(full_path, "video")
+                for av_frame in av_reader.decode(av_reader.streams.video[0]):
+                    vr_frame = next(video_reader)
+
+                    self.assertAlmostEqual(
+                        float(av_frame.pts * av_frame.time_base),
+                        vr_frame["pts"],
+                        delta=0.1,
+                    )
+
+                    av_array = torch.tensor(av_frame.to_rgb().to_ndarray()).permute(
+                        2, 0, 1
+                    )
+                    vr_array = vr_frame["data"]
+                    mean_delta = torch.mean(
+                        torch.abs(av_array.float() - vr_array.float())
+                    )
+                    # on average the difference is very small and caused
+                    # by decoding
+                    # TODO: asses empirically how to set this?
+                    self.assertTrue(mean_delta.item() < 2.5)
+
+            av_reader = av.open(full_path)
+            if av_reader.streams.audio:
+                video_reader = VideoReader(full_path, "audio")
+                for av_frame in av_reader.decode(av_reader.streams.audio[0]):
+                    vr_frame = next(video_reader)
+                    self.assertAlmostEqual(
+                        float(av_frame.pts * av_frame.time_base),
+                        vr_frame["pts"],
+                        delta=0.1,
+                    )
+
+                    av_array = torch.tensor(av_frame.to_ndarray()).permute(1, 0)
+                    vr_array = vr_frame["data"]
+
+                    mean_delta = torch.mean(
+                        torch.abs(av_array.float() - vr_array.float())
+                    )
+                    self.assertTrue(mean_delta.item() < 1)
+
+    def test_metadata(self):
+        """
+        Test that the metadata returned via pyav corresponds to the one returned
+        by the new video decoder API
+        """
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+            reader = VideoReader(full_path, "video")
+            reader_md = reader.get_metadata()
+            self.assertAlmostEqual(
+                config.video_fps, reader_md["video"]["fps"][0], delta=0.0001
+            )
+            self.assertAlmostEqual(
+                config.duration, reader_md["video"]["duration"][0], delta=0.5
+            )
```

Comment:
Isn't this test duplicate compared to `test_predefined_metadata` ?

Reply:
Yes, I was thinking of retiring the `test_video.py` with your permission.

Reply:
ah, ok, there is a duplicate here yes - will remove it.